scripts/test/terra_validation.py

```diff
@@ -0,0 +1,135 @@
+#!/bin/python
+
+import subprocess
+import argparse
+import json
+from os import listdir
+from os.path import basename, isfile
+import logging
+import atexit
+
+
+"""
+Summary: Cursory validation of Terra input JSONs for pipeline workflows and metrics workflows. 
+  Checks if JSONs contain all required inputs and that they do not contain extraneous inputs.
+  Does not perform type-checking. Also checks that all expected JSONs have been generated.
+
+Usage:
+  python scripts/test/terra_validation.py -d /path/to/base/dir -j /path/to/womtool/jar [optional flags]
+
+Parameters:
+  /path/to/base/dir: path to base directory of gatk-sv repo
+  /path/to/womtool/jar: path to user's womtool jar file
+  Optional flags:
+  --log-level LEVEL (specify level of logging information to print, ie. INFO, WARNING, ERROR - not case-sensitive)
+
+Outputs: If successful, last line of printout should read "19 of 19 Terra input JSONs exist and passed validation."
+  Prior lines will detail the JSONs that were examined and any errors found.
+"""
+
+
+TMP_JSON = "womtool_inputs.tmp.json"
+WDLS_PATH = "/wdl/"
+TERRA_INPUTS_PATH = "/inputs/terra_workspaces/cohort_mode/workflow_configurations/"
+METRICS_INPUTS_PATH = "metrics_workflows/"
+NUM_TERRA_INPUT_JSONS = 12
+NUM_TERRA_METRICS_JSONS = 7
+
+
+def exit_handler():
+  if isfile(TMP_JSON):
+    subprocess.run("rm " + TMP_JSON, shell=True)
+
+
+def list_jsons(inputs_path):
+  jsons = [x for x in listdir(inputs_path) if x.endswith(".json")]
+  num_input_jsons = len(jsons)
+  if num_input_jsons < NUM_TERRA_INPUT_JSONS:
+    logging.warning(f"Expected {NUM_TERRA_INPUT_JSONS} Terra input JSONs but found {num_input_jsons}.")
+  jsons.sort()
+
+  metrics_jsons = [METRICS_INPUTS_PATH + x for x in listdir(inputs_path + METRICS_INPUTS_PATH) if x.endswith(".json")]
+  num_metrics_jsons = len(metrics_jsons)
+  if num_metrics_jsons < NUM_TERRA_METRICS_JSONS:
+    logging.warning(f"Expected {NUM_TERRA_METRICS_JSONS} Terra metrics input JSONs but found {num_metrics_jsons}.")
+  metrics_jsons.sort()
+
+  jsons.extend(metrics_jsons)
+  return jsons
+
+
+def get_wdl_json_pairs(wdl_path, terra_inputs_path):
+  jsons = list_jsons(terra_inputs_path)
+
+  for json_file in jsons:
+    path_to_wdl = wdl_path + basename(json_file)[:-5] + ".wdl"
+    if isfile(path_to_wdl):
+      yield path_to_wdl, terra_inputs_path + json_file
+    else:
+      logging.warning(f"Can't find WDL corresponding to {basename(json_file)} at {path_to_wdl}.")
+
+
+def validate_terra_json(wdl, terra_json, womtool_jar):
+  womtool_input_generation_command = "java -jar " + womtool_jar + " inputs " + wdl + " > " + TMP_JSON
+  subprocess.run(womtool_input_generation_command, shell=True)
```

Comment:
You can use `capture_output=True` to load stdout directly into memory as a string rather than writing to a temp file.